core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts

```diff
@@ -51,7 +51,7 @@ export class ExplorationFooterComponent {
   }
 
   ngOnInit(): void {
-    if (this.contextService.getExplorationId()) {
+    try {
```

Comment:
```This component is used at `exploration-player-page` and `practice-session-page` with different usage at both places. The author profiles section exist for former one and the URL contains a valid explorationId while it lacks for the latter. Try-catch here is for catching the error thrown from context-service at getExplorationId so that the component behaves properly at both the places.```

Reply:
done, please suggest any changes you may prefer

Reply:
But why we needed to replace the `if` with `try`?

Reply:
`getExplorationId()` is supposed to be used inside the context of an exploration or a question and no where else. It is used correctly everywhere except at `core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts`. The reason this was moved is because at all the other places getExplorationId was being called, we would have had to put a check for it when it returns `undefined` which leads to very redundant code which can be avoided.

when we used `if` and called `this.contextService.getExplorationId()`. It was throwing an error in the console when used at `practice-session-page` . we expect this behavior to happen as explained in the comment. This was leading the e2e tests to fail since the console should have no errors.

Reply:
Thanks for a thorough explanation, I think this is a bit of codesmell, do you have any idea for a better long term refactor? If yes, could you add a TODO for how to handle it in the future? Maybe create different footer for the practice page?

Reply:
done, created an issue #13494